astrosat/views.py

```diff
@@ -246,6 +253,21 @@ class DatabaseLogRecordViewSet(viewsets.ReadOnlyModelViewSet):
     filterset_class = DatabaseLogRecordFilterSet
 
 
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def feature_tracking(request):
+    """
+    Track Feature usage, so that when a request is received, the code iterates
+    through the list of JSON objects and logs each to the Database Logger.
+    """
+    for record in request.data:
+        logger.info(
+            json.dumps(record['content']), extra={"tags": record['tags']}
+        )
+
```

Comment:
any error-handling in case the record is not JSON-serializable?

Reply:
or, indeed, if it is not a `list`?

Reply:
arggh, you already mentioned this as well, will add `isinstance(request.data, list)` check and return error response if not.

Reply:
See my comment  below about returning an object rather than a string.

Reply:
Also it still doesn't check if it's valid JSON.  A try/catch block would handle that.  Although, as with all my latest comments, this won't stop me from approving it  Since if you do get a JSON error, you'll return a 500 anyway.

Reply:
Cool, I'm fine doing that, I would probably do this after first checking that the POST'd data is an array, do you agree? e.g.

```JavaScript
    if isinstance(request.data, list):
        try:
            json.loads(request.data)
        except:
            return Response(
                    "Request does not contain valid JSON",
                    status=status.HTTP_400_BAD_REQUEST
                )

        for record in request.data:
            if "content" not in record:
                return Response(
                    "Log Record must contain key 'content' of JSON to be logged",
                    status=status.HTTP_400_BAD_REQUEST
                )

            logger.info(
                json.dumps(record['content']), extra={"tags": record['tags']}
            )

        return Response(status=status.HTTP_201_CREATED)
```

Reply:
Functionally, this works.  But it could be made more efficient (you call `json.loads` and don't do anything w/ the result).  Also, there's no information about _what_ went wrong if the JSON is invalid.  I would perhaps write something like this:

```
try:
  assert isinstance(request.data, list), "Must supply an array of JSON objects in request"
  for record in request.data:
    assert "content" in record, "Log Record must contain key 'content' of JSON to be logged"
    logger.info(json.dumps(record["content"]), extra={"tags": record.get("tags"))
except Exception as e:
  return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST
return Response(whatever, status=status.HTTP_201_CREATED)
```
The differences are:
* I use asserts w/ a custom error message which will get picked up if False in the `except` branch
* I assume that any errors w/ `json.loads` will also get picked up in the `except` branch
* I use a `get` for each record's tag instead of directly accessing the key (in case it doesn't exist)

Reply:
aaahhhh, live and learn, cheers. Hope this didn't take up too much of your time.... but it is really good for me.